elfi/examples/stochastic_volatility_model.py

```diff
@@ -65,31 +132,45 @@ def alpha_stochastic_volatility_model(alpha,
         Observations of an alpha-stable SVM.
 
     """
-    random_state = random_state or np.random
-
-    # currently assumes remaining parameters are known and fixed
-    mu = 5
-    phi = 1
-    kappa = 1
-    eta = 0
-    sigma = 0.2
-
-    y_mat = np.zeros((batch_size, n_obs))
-    # first time step (does not rely on prev xx_t)
-    v_0 = shock_term(alpha, beta, kappa, eta, batch_size, random_state)
-    x_0 = random_state.normal(mu+phi*-mu, sigma, batch_size)
-    y_mat[:, 0] = x_0*v_0  # assumes x_0 has no prev.
-    x_prev = x_0
-    for t in range(1, n_obs):
-        # draw log volatility term (x_t)
-        x_t = random_state.normal(mu+phi*(x_prev-mu), sigma, batch_size)
-        # draw shock term from stable distribution
-        v_t = shock_term(alpha, beta, kappa, eta, batch_size, random_state)
-        y_mat[:, t] = x_t * v_t
-        x_prev = x_t
-    if batch_size == 1:
-        y_mat = y_mat.flatten()
-    return y_mat
+    # draw log volatility term
+    x_t = log_vol(mu, phi, sigma, n_obs, x_0, batch_size, random_state)
+    # draw shock term from stable distribution
+    v_t = shock_term(alpha, beta, kappa, eta, n_obs, batch_size, random_state)
+    # calculate returns
+    y_mat = np.exp(0.5 * x_t) * v_t
+    return np.transpose(y_mat)
+
+
+def kurt(x):
+    """Calculate quantile-based kurtosis measure.
+
+    Parameters
+    ----------
+    x : np.array in shape (batch_size, n_obs)
+
+    Returns
+    -------
+    np.array in shape (batch_size, 1)
+
+    """
+    qs = np.quantile(x, q=[0.05, 0.25, 0.75, 0.95], axis=1)
+    return ((qs[3] - qs[0])/(qs[2] - qs[1])).reshape(-1,1)
+
+
+def skew(x):
+    """Calculate quantile-based skewness measure.
+
+    Parameters
+    ----------
+    x : np.array in shape (batch_size, n_obs)
+
+    Returns
+    -------
+    np.array in shape (batch_size, 1)
+
+    """
+    qs = np.quantile(x, q=[0.05, 0.25, 0.50, 0.75, 0.95], axis=1)
+    return (((qs[4] - qs[2]) - (qs[2]) - qs[0])/(qs[3] - qs[1])).reshape(-1,1)
```

Comment:
Isn't this `(((qs[4] + qs[0])/(qs[3] - qs[1])).reshape(-1,1)` ?

Reply:
Can't we eliminate `qs[2]`s here as well from the numerator?

Reply:
no unless i still have an error in the expression. the idea is to compare the absolute distance between the median and 95% quantile `qs[4] - qs[2]` and the absolute distance between the median and the 5% quantile `qs[2] - qs[0]`. we could write the numerator as `qs[4] + qs[0] - 2*qs[2]` but cannot eliminate `qs[2]`.